gdal/ogr/ogrsf_frmts/mysql/ogrmysqldatasource.cpp

```diff
@@ -1099,77 +1189,8 @@ OGRMySQLDataSource::ICreateLayer( const char * pszLayerNameIn,
         mysql_free_result( hResult );
     hResult = nullptr;
 
-/* -------------------------------------------------------------------- */
-/*      Sometimes there is an old crufty entry in the geometry_columns  */
-/*      table if things were not properly cleaned up before.  We make   */
-/*      an effort to clean out such cruft.                              */
-/*                                                                      */
-/* -------------------------------------------------------------------- */
-    osCommand.Printf(
-             "DELETE FROM geometry_columns WHERE f_table_name = '%s'",
-             pszLayerName );
-
-    if( mysql_query(GetConn(), osCommand ) )
-    {
-        ReportError( osCommand );
+    if( UpdateMetadataTables(pszLayerName, eType, pszGeomColumnName, nSRSId) != OGRERR_NONE )
```

Comment:
Moving these chunks and introduce UnpdateMetadataTables() makes ICreateLayer()  simpler and easy to understand by combination with InitializeMetadataTables(), which are necessary only when legacy MySQL and MariaDB.